.github/workflows/release-against-rancher.yml

```diff
@@ -53,8 +54,7 @@ jobs:
           path: rancher
       - uses: actions/setup-go@v4
         with:
-          go-version-file: 'go.mod'
-          check-latest: true
+          go-version: ${{ env.SETUP_GO_VERSION }}
```

Comment:
By curiousity
What's the motivation to explicitly specify the golang version here? 
Instead of reusing the one specified in the go.mod file? 

From my perspective, it's one additional dependency that we need to monitor